### PR TITLE
Add support to gormv2

### PR DESCRIPTION
**Pull request by tscolari**

### CODE REVIEW

File: pkg/ocgormv2/context.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+var (
+	contextScopeKey = "_opencensusContext"
+)
+
+// WithContext sets the current context in the db instance for instrumentation.
+func WithContext(ctx context.Context, db *gorm.DB) *gorm.DB {
+	// we MUST call `Session` here because otherwise resulting *gorm.DB will be unsafe
+	// to reuse, see https://gorm.io/docs/method_chaining.html for more information
+	return db.Set(contextScopeKey, ctx).Session(&gorm.Session{})
```
Comment (Dombo) [inline pkg/ocgormv2/context.go:17]: Doesn't this move all queries into a transaction? Just reading through the [.Session](https://github.com/go-gorm/gorm/blob/v1.25.4/gorm.go#L218) method.

Comment (Dombo) [reply pkg/ocgormv2/context.go:17]: Ahh WithContext already does that, so this should represent no change for consumers. I agree with this patch.

**Comment by smaant** [comment]
@Dombo converted it into a ready for review PR, let me know if I should go ahead and merge it, thank you

**Comment by Dombo** [comment]
Closed in favour of #11
Thanks @dmalch-hashicorp !

**Pull request closed without merging**